### PR TITLE
Clean up header file defines and comments to make it more consistent

### PR DESCRIPTION
**Pull request by chutsu**
In relation to #128 

This PR just makes the header file #defines and its comments more consistent

#### Pre-Merge Checklist:
- [x] Code is documented in doxygen format
- [x] Code has automated tests
- [x] Zero compiler warnings
- [x] Formatted with `clang-format`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,18 +18,18 @@ Execute the following in the terminal where you want libwave to reside:
 
 **Or** you can perform the installation manually:
 
-    # clone repo
+#clone repo
```
Comment (leokoppel) [inline README.md]: This makes it a top-level header, instead of a comment in a code block?

Comment (chutsu) [inline README.md]: Opps I was playing around testing clang-format

Comment (chutsu) [inline README.md]: Fixed

**Comment by leokoppel** [comment]
Should do .cpp files as well. 
I'm going to jump in and replace "end of group" in the Doxygen comment to something similar, e.g. "group vision".

File: wave_kinematics/include/wave/kinematics/quadrotor.hpp [hunk 2]
```diff
@@ -136,5 +136,6 @@ class QuadrotorModel {
 };
 
 /** @} end of group */
-}  // end of wave namespace
-#endif
+}  // namespace wave
+
+#endif  // WAVE_QUADROTOR_QUADROTOR_HPP
```
Comment (navganti) [inline wave_kinematics/include/wave/kinematics/quadrotor.hpp]: WAVE_KINEMATICS_QUADROTOR

File: wave_matching/include/wave/matching/pcl_common.hpp [hunk 3]
```diff
@@ -22,5 +22,6 @@ namespace wave {
 typedef pcl::PointCloud<pcl::PointXYZ>::Ptr PCLPointCloud;
 
 /** @} end of group */
-}  // end of namespace wave
+}  // namespace wave
+
 #endif  // WAVE_PCL_COMMON_HPP
```
Comment (navganti) [inline wave_matching/include/wave/matching/pcl_common.hpp]: should this have WAVE_MATCHING_...?

File: wave_vision/include/wave/vision/detector/fast_detector.hpp [hunk 4]
```diff
@@ -124,6 +120,6 @@ class FASTDetector : public FeatureDetector {
 };
 
 /** @} end of group */
-}  // end of namespace wave
+}  // namespace wave
 
 #endif  // WAVE_FAST_DETECTOR_HPP
```
Comment (navganti) [inline wave_vision/include/wave/vision/detector/fast_detector.hpp]: WAVE_VISION_FAST....

(sorry, my bad)

File: wave_matching/include/wave/matching/pcl_common.hpp (hunk 3 above)
Comment (chutsu) [inline wave_matching/include/wave/matching/pcl_common.hpp]: yes

File: wave_kinematics/include/wave/kinematics/quadrotor.hpp (hunk 2 above)
Comment (chutsu) [inline wave_kinematics/include/wave/kinematics/quadrotor.hpp]: Done

File: wave_matching/include/wave/matching/pcl_common.hpp (hunk 3 above)
Comment (chutsu) [inline wave_matching/include/wave/matching/pcl_common.hpp]: Done

File: wave_vision/include/wave/vision/detector/fast_detector.hpp (hunk 4 above)
Comment (chutsu) [inline wave_vision/include/wave/vision/detector/fast_detector.hpp]: Its aite, small stuff.

File: wave_geometry/include/wave/geometry/numerical_test_functors.hpp [hunk 5]
```diff
@@ -2,8 +2,8 @@
  * @ingroup geometry
  */
 
-#ifndef WAVE_NUMERICAL_TEST_FUNCTORS_HPP
-#define WAVE_NUMERICAL_TEST_FUNCTORS_HPP
+#ifndef WAVE_GEOMOETRY_NUMERICAL_TEST_FUNCTORS_HPP
```
Comment (navganti) [inline wave_geometry/include/wave/geometry/numerical_test_functors.hpp]: typo in geometry

Comment (chutsu) [inline wave_geometry/include/wave/geometry/numerical_test_functors.hpp]: done

**Pull request merged**